### PR TITLE
Cover the chunked SAM3 mask upscaling with a regression test

### PR DESCRIPTION
**Pull request by rahultechenable**
Follows #26196, which chunks the SAM3 semantic mask upscale to bound peak memory (#26194).

This is stacked on `fix/sam3-mask-upscale-memory` and targets that branch rather than `main`, because `_upscale_masks` only exists there. Happy to retarget to `main` once #26196 merges, or for the two hunks to just be folded into #26196 if you'd rather keep it in one place.

### Why the constant

The chunk size in `_upscale_masks` is currently an inline literal:

```python
chunk = max(1, 2**29 // (shape[0] * shape[1]))
```

Reaching `chunk < N` requires `H*W > 2**29 / N` — about 16384x16384 even at 8 masks, whose bool output alone is ~2.1 GB. Every CI-sized shape therefore collapses to a single chunk, so a test would silently exercise the old single-shot path instead of the new one. That is what the 11% patch coverage on #26196 reflects.

Hoisting the budget to a module-level `MAX_UPSCALE_ELEMENTS` lets a test shrink it, mirroring how `ops.process_mask_native` is covered by the existing `test_process_mask_native_chunked`.

### The test

`test_sam3_upscale_masks_chunked` asserts two things:

1. the chunked result is identical to a single-shot upscale, and
2. more than one chunk is actually processed, by counting `F.interpolate` calls.

The second assertion is the one that earns its place. Without it, removing the chunking altogether still passes, because a single-shot implementation trivially satisfies "chunked == single-shot". Both were verified by mutation: inverting the threshold comparison and replacing the chunked loop with a single upscale each fail the test as written.

Runs in ~0.05s on CPU. No weights, no GPU, no network.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds regression coverage for chunked SAM3 semantic mask upscaling and exposes its element budget as a module-level constant for deterministic testing.

### 📊 Key Changes
- Introduces `MAX_UPSCALE_ELEMENTS` in `ultralytics.models.sam.predict` and uses it to calculate mask upscale chunk sizes.
- Adds `test_sam3_upscale_masks_chunked`, which forces multiple chunks, counts interpolation calls, verifies boolean output, and compares results with single-shot interpolation.
- Confirms the chunked path processes seven masks in four chunks when the test budget is reduced to two output masks per chunk.

### 🎯 Purpose & Impact
- Ensures SAM3 mask upscaling continues to use bounded chunks while preserving single-shot results. No user-facing change with the default budget.

### CODE REVIEW

**Comment by rahultechenable** [comment]
I have read the CLA Document and I sign the CLA

**Comment by UltralyticsAssistant** [comment]
👋 Hello @rahultechenable, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms your contribution was received, and an Ultralytics engineer will assist with the review. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Reviewed both changed files, including the SAM3 upscaling implementation, callers, and regression test. The constant extraction preserves behavior, and the test verifies both chunk equivalence and that multiple interpolation calls occur. LGTM.
